src/test/java/zk_proof_systems/zkSNARK/grothBGM17/ZKSnarkObjectWriterTest.java

```diff
@@ -0,0 +1,86 @@
+package zk_proof_systems.zkSNARK.grothBGM17;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import algebra.curves.AbstractG1;
+import algebra.curves.AbstractG2;
+import algebra.curves.barreto_naehrig.bn254a.BN254aBinaryReader;
+import algebra.curves.barreto_naehrig.bn254a.BN254aBinaryWriter;
+import algebra.curves.barreto_naehrig.bn254a.BN254aFields.BN254aFr;
+import algebra.curves.barreto_naehrig.bn254a.BN254aG1;
+import algebra.curves.barreto_naehrig.bn254a.BN254aG2;
+import algebra.curves.barreto_naehrig.bn254a.bn254a_parameters.BN254aG1Parameters;
+import algebra.curves.barreto_naehrig.bn254a.bn254a_parameters.BN254aG2Parameters;
+import algebra.fields.AbstractFieldElementExpanded;
+import io.BinaryCurveReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import org.junit.jupiter.api.Test;
+import zk_proof_systems.zkSNARK.grothBGM17.objects.Proof;
+
+class ZKSnarkObjectWriterTest {
+
+  protected <
+          FrT extends AbstractFieldElementExpanded<FrT>,
+          G1T extends AbstractG1<G1T>,
+          G2T extends AbstractG2<G2T>>
+      Proof<G1T, G2T> dummyProof(final FrT oneFr, final G1T oneG1, final G2T oneG2) {
+    return new Proof<G1T, G2T>(
+        oneG1.mul(oneFr.construct(-3)),
+        oneG2.mul(oneFr.construct(-4)),
+        oneG1.mul(oneFr.construct(-5)));
+  }
+
+  protected <
+          FrT extends AbstractFieldElementExpanded<FrT>,
+          G1T extends AbstractG1<G1T>,
+          G2T extends AbstractG2<G2T>>
+      void writeProof(
+          final ZKSnarkObjectWriter<FrT, G1T, G2T> zkSnarkObjectWriter,
+          final FrT oneFr,
+          final G1T oneG1,
+          final G2T oneG2)
+          throws IOException {
+    final var dummyProof =
```

Comment:
Why not use the function here. i.e. `dummyProof(oneFr, oneG1, oneG2),`?

Reply:
Switched to using the function.